x-pack/plugins/security_solution/common/api/timeline/import_timelines/import_timelines_route.schema.yaml

```diff
@@ -28,37 +28,21 @@ paths:
           application/json:
             schema:
               type: object
+              required: [file]
               properties:
-                file:
-                  allOf:
-                    - $ref: '../model/components.schema.yaml#/components/schemas/Readable'
-                    - type: object
-                      required: [hapi]
-                      properties:
-                        hapi:
-                          type: object
-                          required: [filename, headers]
-                          properties:
-                            filename:
-                              type: string
-                            headers:
-                              type: object
-                            isImmutable:
-                              type: string
-                              enum:
-                                - 'true'
-                                - 'false'
+                isImmutable:
+                  type: string
+                  enum:
+                    - 'true'
+                    - 'false'
+                file: {}
```

Comment:
This one is tricky, since `file` is not something that needs to be provided by the API. I tried to model this as a [file upload](https://swagger.io/docs/specification/describing-request-body/file-upload/) but the generator script didn't manage to generate schemas for these. (cc @maximpn)

Reply:
Import timelines route's implementation is similar to import rules. The latter [has](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/rule_management/import_rules/import_rules_route.schema.yaml) `multipart/form-data` MIME type with binary format. It looks like you need the same here.

Router's implementation isn't able to validate file uploads (Even if it could it'd require uploading the whole file disabling streaming processing). It means that defining file schema in OpenAPI isn't required. It's up to the route's implementation to perform proper validations. It's always possible to define necessary Zod schemas manually for that purpose and replace `io-ts` counterparts.